controllers/providers/aws/ec2.go

```diff
@@ -62,16 +63,25 @@ func (a *AmazonClientSet) DescribeTaggedInstanceIDs(tagKey, tagValue string) ([]
 	return instances, err
 }
 
-func (a *AmazonClientSet) TagEC2instance(instanceID, tagKey, tagValue string) error {
-	input := &ec2.CreateTagsInput{
-		Resources: aws.StringSlice([]string{instanceID}),
-		Tags: []*ec2.Tag{
-			{
-				Key:   aws.String(tagKey),
-				Value: aws.String(tagValue),
+func (a *AmazonClientSet) TagEC2instance(instances []*autoscaling.Instance, tagKey, tagValue string) error {
+	var instanceIDs []string
+	for _, instance := range instances {
+		if aws.StringValue(instance.LifecycleState) == autoscaling.LifecycleStateInService {
+			instanceIDs = append(instanceIDs, aws.StringValue(instance.InstanceId))
```

Comment:
Why do we only want to tag InService instances? you can tag instances regardless of their state, no?

Reply:
The possible states the instance can be when it arrives at this point in code are - `InService, Pending, Detaching, StandBy`. Instances that are in `StandBy` already have in-progress tag.  By this check, we can avoid the additional AWS calls to tag instances repeatedly or avoid tagging the instances that we aren't ready to operate on.

Reply:
Yeah, but it doesn't make sense from code perspective, because the method `TagEC2instance` is supposed to be generic, if we want to do any filtering based on lifecycle-state it should happen on the controller side e.g. only pass InService instances to TagEC2Instance - you should also rename it to `TagEC2Instances`.

Reply:
Hmm. That's right. I can move the checking to controller code.